### PR TITLE
Remove sealing 

### LINKED ISSUE #51: Catastophic recovery using key shares
Use https://github.com/dsprenkels/sss adapted to run in an enclave to implement Shamir's Secret Sharing.

- [x] 1. Link sss to CCF enclave (#810)
- [x] 2. Add `ksk_m` to `ccf.members` table and generate in test infra/docs.
- [x] 3. Generate `k_z` on start-up, split it into "some" shares, encrypt the shares and keep encrypted shares in node's memory for now.
- [x] 4. Add new `ccf.shares` table and populate it with `(ledger secrets)k_z` and `[(shares_m)ksk_m]` on start-up.
- [x] 5. Serve members' shares on `getEncryptedRecoveryShare` RPC and allow members to easily decrypt them.
- [x] 6. Pass share to new `submitRecoveryShare` RPC (after vote).
- [x] 6'. Re-construct `k_z` when sufficient shares (i.e. secret sharing threshold) have been submitted, decrypt ledger secrets from `ccf.shares` table and initiate private ledger recovery.
- [x] 6''. Support for multiple recoveries.
**[When all the steps above are implemented, this should be enough to have a functioning, yet incomplete, recovery scheme]**
- [x] 7. Generate ~_endorsed_~ network encryption key `nk` and pass it via the join protocol. The public key should be output by the starting node.
- [x] ~8. Members should encrypt their shares with `nk_pub` before passing it to their vote RPC. CCF should decrypt the encrypted share on proposal completion.~
- [ ] 9. Resilience to elections: submitted shares should be stored in the KV + end-to-end test for election during recovery (will be addressed in https://github.com/microsoft/CCF/issues/1051). 
- [x] 10. Issue new member shares on consortium update.
- [x] 11. Issue new member shares on re-keying (to be addressed in https://github.com/microsoft/CCF/issues/1049).
- [ ] 12. Remove sealing and update docs.
- [x] 13. ~Always re-key the ledger when issuing new shares.~ Re-key ledger when retiring active member.
- [x] 14. Threshold should be defined in a new configuration table and updatable via governance.

**Misc*

- [ ] `ServiceStatus` should use a state machine, in a similar way to `NodeState`.

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #51 

- Remove `Seal` class and code that was using it.
- Python infra now only tests recovery with key shares.
- Documentation updated accordingly.

### CODE REVIEW

File: sphinx/source/members/accept_recovery.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+When the recovery threshold is reached, the ``submitRecoveryShare`` RPC returns ``true``. At this point, the private recovery procedure is started and the private ledger is being recovered.
+
+.. note:: While all nodes are recovering the private ledger, no new transaction can be executed by the network.
+
+Once the recovery of the private ledger is complete on a quorum of nodes that have joined the new network, the ledger is fully recovered and users are able to continue issuing business transactions.
+
+.. note:: Recovery shares are updated every time a new member is added or retired and when the ledger is rekeyed. It also possible for members to update the recovery shares via the ``update_recovery_shares`` proposal.
+
+Summary Diagram
```
Comment (jumaffre) [inline sphinx/source/members/accept_recovery.rst:72]: image

File: sphinx/source/members/adding_member.rst [hunk 2]
```diff
@@ -3,25 +3,27 @@ Adding New Members
 
 It is possible for existing members to add new members to the consortium after a CCF network has been started.
 
+.. note:: The maximum number of active consortium members at any given time is 255.
```
Comment (jumaffre) [inline sphinx/source/members/adding_member.rst:6]: Not closing https://github.com/microsoft/CCF/issues/851 just yet. I will add some checks in CCF to make sure that the 256th member `ACK` fails.

File: sphinx/source/members/accept_recovery.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+========================================
+
+.. note:: Before members can initiate the end of the recovery procedure, operators should have started a new network and recovered all public transactions. See :ref:`details for public recovery operator procedure <operators/recovery:Establishing a Recovered Public Network>`.
+
+Accepting Recovery
+------------------
+
+Once the public recovered network has been established by operators, members are allowed to vote to confirm that the configuration of the new network is suitable to complete the recovery procedure.
+
+The first member proposes to recover the network:
```
Comment (olgavrou) [inline sphinx/source/members/accept_recovery.rst:11]: Maybe "a member proposes to recover the network" instead. `The first member` implies a specific member has to propose the network recovery. 

Also add: "and other members can vote on the proposal" before the voting example?

File: sphinx/source/operators/recovery.rst [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
-.. note:: The current version of the recovery protocol relies on Intel SGX Sealing capability [#sealing]_.
+The recovery procedure consists of two phases:
 
-However, one of the previous ledgers can be recovered and the execution of new business transactions continue if the following three conditions are met:
+1. Operators should retrieve one of the ledgers of the previous service and re-start one or several nodes in ``recover`` mode. The public transactions of the previous network are restored and the new network established.
 
-- At least one of the old nodes' CPU survived.
-- The sealed network secret file (``sealed_secrets.<date>.<pid>``) associated with that CPU is available to enough members, where enough is a set of members that can pass a recovery proposal under the rules of the :term:`constitution`.
-- One of the ledgers (preferably the ledger of the previous primary as it is likely to be the longest) is available.
+2. After agreeing that the configuration of the new network is suitable, members should vote to accept to recover the network and once this is done, submit their recovery shares to initiate the end of the recovery procedure.
```
Comment (olgavrou) [inline sphinx/source/operators/recovery.rst:10]: maybe link to the section describing how members submit their recovery shares

File: sphinx/source/members/accept_recovery.rst (hunk 3 above)
Comment (jumaffre) [reply sphinx/source/members/accept_recovery.rst]: Good point! Fixed now.

File: sphinx/source/members/accept_recovery.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+Once the proposal to recover the network has passed under the rules of the :term:`constitution`, the recovered service is ready for members to submit their recovery shares.
+
+Submitting Recovery Shares
+--------------------------
+
+To restore private transactions and complete the recovery procedure, members shoud submit their recovery shares. The number of members required to submit their shares is set by the ``recovery_threshold`` CCF configuration parameter and :ref:`can be updated by the consortium at any time <members/common_member_operations:Updating Recovery Threshold>`.
+
+.. note:: The members who submit their recovery shares do not necessarily have to be the members who previously accepted the recovery.
+
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply sphinx/source/members/accept_recovery.rst:42]: Absolutely. This ties into https://github.com/microsoft/CCF/issues/1096 but we need to find/write a good CLI utility to perform cryptobox decryption first. I will update this docs accordingly when this is addressed.

**Pull request merged**